store/sqlstore/channel_store.go

```diff
@@ -2225,10 +2254,27 @@ func (s SqlChannelStore) UpdateLastViewedAtPost(unreadPost *model.Post, userID s
 	if err = s.GetMaster().SelectOne(result, chanUnreadQuery, params); err != nil {
 		return nil, errors.Wrapf(err, "failed to get ChannelMember with channelId=%s", unreadPost.ChannelId)
 	}
+
+	go s.Thread().UpdateUnreadsByChannel(userID, threadsToUpdate, result.LastViewedAt)
 	return result, nil
 }
 
-func (s SqlChannelStore) IncrementMentionCount(channelId string, userId string) error {
+func (s SqlChannelStore) IncrementMentionCount(channelId string, userId string, updateThreads bool) error {
+	var threadsToUpdate []string
+	if updateThreads {
+		var err error
+		timestamp, err := s.GetReplica().SelectInt("SELECT LastUpdateAt FROM ChannelMembers WHERE UserId = :UserId AND ChannelId = :ChannelId", map[string]interface{}{"ChannelId": channelId, "UserId": userId})
```

Comment:
This timestmap is not needed, that is get internally in the database, the one that you need is the `now`, so you can define the `now` up in the function, and use that now here, in the update of the channelMembership and in the update of the threads.

Reply:
@jespino - we are using 'now' after the update. Before the update - we want the actual timestamp I think

Reply:
The actual timestamp is get as part of the `ChannelMembers.LastViewAt` in the query inside.

Reply:
@jespino that's LastViewAt, here I am fetching LastUpdateAt and it's used for comparison:
```go
query, args, _ := s.getQueryBuilder().
		Select("Threads.PostId").
		From("Threads").
		LeftJoin("Posts ON Threads.PostId=Posts.Id").
		LeftJoin("ChannelMembers ON ChannelMembers.ChannelId=Posts.ChannelId").
		Where(sq.And{
			sq.Eq{"Posts.ChannelId": channelIds},
			sq.Eq{"ChannelMembers.UserId": userId},
			sq.Or{
				sq.Expr("Threads.LastReplyAt >= ChannelMembers.LastViewedAt"),
				sq.GtOrEq{"Threads.LastReplyAt": timestamp},
			},
		}).
		ToSql()
```